internal/service/console.go

```diff
@@ -0,0 +1,40 @@
+package service
+
+import (
+	"context"
+
+	"github.com/flightctl/flightctl/internal/api/server"
+	"github.com/flightctl/flightctl/internal/flterrors"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/google/uuid"
+)
+
+func (h *ServiceHandler) RequestConsole(ctx context.Context, request server.RequestConsoleRequestObject) (server.RequestConsoleResponseObject, error) {
+	orgId := store.NullOrgId
+
+	// make sure the device exists
+	_, err := h.store.Device().Get(ctx, orgId, request.Name)
+	if err != nil {
+		switch err {
+		case flterrors.ErrResourceNotFound:
+			return server.RequestConsole404JSONResponse{}, nil
+		default:
+			return nil, err
+		}
+	}
+
+	sessionId := uuid.New().String()
+
+	annotations := map[string]string{"flightctl.io/console": sessionId}
+
+	if err := h.store.Device().UpdateAnnotations(ctx, orgId, request.Name, annotations, []string{}); err != nil {
```

Comment:
How does the user stop the console connection?  We currently don't allow setting/deleting annotations.  We should either add an API to stop the connection or allow modifying annotations.

Reply:
Yes, this is a POC now, but I am going to open a RFE in this PR with the next items, this should be one of those.

In the current code, once the device or the client disconnects, the server will prevent new connections through the same session ID. Also, the device does not connect if it's a session ID it connected to and disconnected. But that's just a simple version for now, we need to clear those annotations when a session ends from one side (client disconnected or process on the device exited).